integrations/docker/images/chip-build-zap/Dockerfile

```diff
@@ -1,25 +1,44 @@
-FROM alpine:3.14
-
-RUN apk add --no-cache \
-  nodejs=14.19.0-r0 \
-  openjdk11=11.0.14_p9-r0 \
-  npm=7.17.0-r0 \
-  python3=3.9.5-r2 \
-  pixman-dev=0.40.0-r2 \
-  cairo-dev=1.16.0-r3 \
-  pango-dev=1.48.5-r0 \
-  make=4.3-r0 \
-  g++=10.3.1_git20210424-r2 \
-  libjpeg-turbo-dev=2.1.0-r0 \
-  giflib-dev=5.2.1-r0 \
-  git=2.32.0-r0 \
-  clang-dev=11.1.0-r1 \
-  py3-pip=20.3.4-r1
-
-# Generator dependencies
-RUN pip3 install --no-cache-dir \
-    click==8.0.3                \
-    coloredlogs==15.0.1         \
-    jinja2==3.0.3               \
-    lark==1.0.0                 \
-    stringcase==1.2.0
+ARG VERSION=latest
+FROM connectedhomeip/chip-build:${VERSION}
+
+ENV ZAP_COMMIT=7ab717d08dfe9b0ba9de907fc7c6eb6549c86bf7
+
+ENV CHIP_ORIGIN=https://github.com/project-chip/connectedhomeip
+ENV ZAP_ORIGIN=https://github.com/project-chip/zap.git
+
+RUN echo "Must be built in a CHIP repo with the included build script and on a commit found in $CHIP_ORIGIN"
+
+ENV LOCAL_COMMIT_CACHE_FILE=/local_chip_commit.txt
+ENV REMOTE_ZAP_CACHE_FILE=/remote_zap_commit.txt
+ENV REMOTE_DOCKERFILE_CACHE_FILE=/remote_dockerfile_commit.txt
+ENV VERIFY_SCRIPT=zapverify.sh
+
+# Check zap in chip origin
+RUN echo 'If you receive a "file not found in build context" error, ensure you are building with the included build.sh to populate the build context'
+COPY context.txt $LOCAL_COMMIT_CACHE_FILE
+WORKDIR /tmp
+RUN git clone $CHIP_ORIGIN
```

Comment:
should we checkout a specific version of CHIP so that the zap version is consistent?

I am not suggesting a specific path here, just wondering what we can do to have a "same docker file always builds the same image" and whenever I see a git clone I see that as a moving target.